src/bin/bobd.rs

```diff
@@ -1,23 +1,61 @@
-use bob::api::grpc::{server, PutRequest, GetRequest, OpStatus, BlobResult};
+use bob::api::grpc::{server, PutRequest, GetRequest, OpStatus, BlobResult, Null};
 
 use futures::{future, Future, Stream};
+use futures::future::ok;
 use tokio::executor::DefaultExecutor;
 use tokio::net::TcpListener;
 use tower_h2::Server;
 use tower_grpc::{Request, Response};
+use bob::core::grinder::{Grinder};
+use bob::core::data::{BobKey, BobData, BobOptions};
+use bob::core::backend::Backend;
+use bob::core::sprinkler::{Sprinkler};
+use tokio::runtime::{Runtime};
+extern crate clap;
 
-#[derive(Clone, Debug)]
-struct BobSrv;
+use clap::{App, Arg};
+
+#[derive(Clone)]
+struct BobSrv {
+    grinder: Grinder
+}
+
+impl BobSrv {
+    pub fn get_periodic_tasks(&self, ex: tokio::runtime::TaskExecutor) -> Box<impl Future<Item=(), Error=()>> {
+        self.grinder.get_periodic_tasks(ex)
+    }
+}
 
 impl server::BobApi for BobSrv {
-        type PutFuture = future::FutureResult<Response<OpStatus>, tower_grpc::Status>;
+        type PutFuture = Box<Future<Item=Response<OpStatus>,Error=tower_grpc::Status> + Send>;
```

Comment:
Why this future is wrapped in `Box<>`?

Reply:
```Future``` is a trait, object that implements it has unknown compile time size and can't be allocated on stack. [```Box```](https://doc.rust-lang.org/std/boxed/struct.Box.html) is a pointer for heap allocation

Reply:
in new version rust it could be changes how declare reference to trait. As alreadyin nightly youcan return trait (fn some() -> impl Future) but this not always work